### PR TITLE
Twap test structure

### PR DESCRIPTION
**Pull request by anxolin**
This PR just restructures the tests.

In my opinion, tests should assert one thing. This will make it easier to understand the scope, maintain, and know exactly what is wrong when tests start to fail.

For this reason, I reorganized the tests (in anticipation to the fact I want to create a new PR with new tests)

## Conditional Order
image

## Twap
image

### CODE REVIEW

File: src/composable/ConditionalOrder.spec.ts [hunk 1]
```diff
@@ -122,3 +141,5 @@ class TestConditionalOrder extends ConditionalOrder<string, string> {
     throw new Error('Method not implemented.')
   }
 }
+
+describe('Constuctor', () => {})
```
Comment (alfetopito) [inline src/composable/ConditionalOrder.spec.ts:145]: Should it have something inside?

File: src/composable/orderTypes/Twap.spec.ts [hunk 2]
```diff
@@ -109,14 +143,30 @@ describe('Twap', () => {
...
       reason: 'InvalidFrequency',
     })
+  })
+
+  test('Invalid twap: InvalidSpan (auto)', () => {
+    expect(Twap.fromData({ ...TWAP_PARAMS_TEST, appData: constants.AddressZero }).isValid()).toEqual({
+      isValid: false,
+      reason: 'InvalidData',
+    })
+  })
```
Comment (alfetopito) [inline src/composable/orderTypes/Twap.spec.ts:167]: This is not an `InvalidSpan` error

File: src/composable/orderTypes/Twap.spec.ts [hunk 3]
```diff
@@ -129,26 +179,26 @@ describe('Twap', () => {
...
   })
 
-  test('isValid: Fails if appData has a wrong number of bytes', () => {
+  test('Invalid twap: InvalidData (ABI parse error in appData)', () => {
     // The isValid below test triggers a throw by trying to ABI parse `appData` as a `bytes32` when
     // it only has 20 bytes (ie. an address)
     expect(Twap.fromData({ ...TWAP_PARAMS_TEST, appData: constants.AddressZero }).isValid()).toEqual({
       isValid: false,
       reason: 'InvalidData',
     })
```
Comment (alfetopito) [inline src/composable/orderTypes/Twap.spec.ts:190]: This test is identical to https://github.com/cowprotocol/cow-sdk/pull/154/files#diff-4b89c7108b87cda7a7ae6bf0bc217684021f6ea46596743c96cfa0b657205d34R162

File: src/composable/orderTypes/Twap.spec.ts [hunk 4]
```diff
@@ -129,26 +179,26 @@ describe('Twap', () => {
...
+})
 
+describe('Serialize', () => {
   test('serialize: Serializes correctly', () => {
     const twap = Twap.fromData(TWAP_PARAMS_TEST)
     expect(twap.serialize()).toEqual(TWAP_SERIALIZED(twap.salt))
   })
+})
 
+describe('Deserializes', () => {
```
Comment (alfetopito) [inline src/composable/orderTypes/Twap.spec.ts:201]:
```suggestion
describe('Deserialize', () => {
```

To keep the same verb tense as `Serialize`

File: src/composable/orderTypes/Twap.spec.ts [hunk 5]
```diff
@@ -157,7 +207,9 @@ describe('Twap', () => {
   test('deserialize: Throws if invalid', () => {
     expect(() => Twap.deserialize('0x')).toThrow('InvalidSerializedConditionalOrder')
   })
+})
 
+describe('To Sting', () => {
```
Comment (alfetopito) [inline src/composable/orderTypes/Twap.spec.ts:212]: Typo?

```suggestion
describe('To String', () => {
```

File: src/composable/orderTypes/Twap.spec.ts (hunk 2 above)
Comment (anxolin) [reply src/composable/orderTypes/Twap.spec.ts]: indeed

File: src/composable/orderTypes/Twap.spec.ts (hunk 3 above)
Comment (anxolin) [reply src/composable/orderTypes/Twap.spec.ts:190]: good catch!

File: src/composable/orderTypes/Twap.spec.ts (hunk 4 above)
Comment (anxolin) [reply src/composable/orderTypes/Twap.spec.ts:194]: corrected

**Comment by anxolin** [comment]
Addressed the comments and low risk PR (also test pass)

I will merge to not waterfall too much

**Pull request merged**